velox/common/memory/SharedArbitrator.cpp

```diff
@@ -315,19 +332,7 @@ void SharedArbitrator::wakeupGlobalArbitrationThread() {
 }
 
 SharedArbitrator::~SharedArbitrator() {
-  shutdownGlobalArbitration();
-
-  VELOX_MEM_LOG(INFO) << "Stopping memory reclaim executor '"
-                      << memoryReclaimExecutor_->getName() << "': threads: "
-                      << memoryReclaimExecutor_->numActiveThreads() << "/"
-                      << memoryReclaimExecutor_->numThreads()
-                      << ", task queue: "
-                      << memoryReclaimExecutor_->getTaskQueueSize();
-  memoryReclaimExecutor_.reset();
-  VELOX_MEM_LOG(INFO) << "Memory reclaim executor stopped";
-
-  VELOX_CHECK_EQ(
-      participants_.size(), 0, "Unexpected alive participants on destruction");
+  shutdown();
```

Comment:
shutdown() is called twice, once MemoryManager and another time here. Do we want to delegate the responsibility to external users of SharedArbitrator to call shudown() and only do a shutdown check here?

Reply:
The memory manager needs to shutdown the arbitrator to stop any background activities to release all the pending memory pool references.